src/datanode/src/sql/create.rs

```diff
@@ -135,6 +135,35 @@ impl SqlHandler {
             .map(|(k, v)| (v, k))
             .collect::<HashMap<_, _>>();
 
+        let pk_map = stmt
+            .columns
+            .iter()
+            .filter(|col| {
+                col.options
+                    .iter()
+                    .any(|options| match options.option.clone() {
+                        ColumnOption::Unique { is_primary } => is_primary,
+                        _ => false,
+                    })
+            })
+            .enumerate()
+            .map(|(_, col)| col.name.value.clone())
+            .collect::<Vec<_>>();
+
+        ensure!(
+            pk_map.len() < 2,
+            InvalidPrimaryKeySnafu {
+                msg: "Multiple definitions of primary key found"
+            }
+        );
+
+        if pk_map.first().is_some() {
+            let pk = pk_map.first().unwrap();
```

Comment:
Consider using `if-let`.
```suggestion
        if let Some(pk) = pk_map.first() {
```

Reply:
fixed